src/sentry_scope.h

```diff
@@ -19,6 +19,12 @@ typedef struct sentry_scope_s {
     sentry_value_t breadcrumbs;
     sentry_level_t level;
     sentry_value_t client_sdk;
+    // Not to be confused with transaction, which is a legacy value. This is
+    // also known as a transaction, but to maintain consistency with other SDKs
+    // and to avoid a conflict with the existing transaction field this is named
+    // span. Whenever possible, `transaction` should pull its value from the
+    // `name` property nested in this field.
+    sentry_value_t span;
```

Comment:
The description is very nice! Maybe name this `root_span` or something? Not sure what would make this less confusing.
Also not sure if you want to save the innermost span somewhere as well?

Reply:
truth be told i'm actually not the biggest fan of calling these spans as there are subtle differences between a transaction and a span. the nice thing about using `span` here is mostly that it's consistent with the naming scheme used by other SDKs (see https://github.com/getsentry/sentry-python/blob/dd0efc08414ee2ef1a5f22d2cc4e243b54a1b455/sentry_sdk/scope.py#L81-L83). 

given that, i'm reluctant to use `root_span` because in theory, this "span" should be a transaction which means it actually should contain some sort of pointer to a span tree(s). there have been suggestions to call this something like `transaction_object`, which makes it less of a misnomer at the cost of giving up consistency. thoughts?

i think the question you have about the innermost span might be indirectly answered by the second paragraph: the stored "span", aka the transaction should have enough info to find any given span that branches off of it if needed. i'll admit however that i don't have a very good vision of how span tree construction and management will look like yet, and i'm leaving that as a challenge to be solved a little later.

Reply:
there's also a setter on scope that the spec explicitly requires to be named `set_span`: https://develop.sentry.dev/sdk/performance/#scope-changes

given that this setter is meant to be internal and this suggested change's intent is to make things easier for us internally, i'd like to do this: i'm going to merge this as-is while it still uses the terminology as required by the spec and as used by all of the other SDKs. a follow-up PR can be opened up which introduces this rename of `span` to `transaction_object` in the scope in addition to the introduction of a `set_transaction_object()` helper to mirror the rename. we can then assess whether the rename works or makes sense or not in there.